compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4647,7 +4647,13 @@ private void visitCheckAndCheckPanicExpr(BLangCheckedExpr checkedExpr) {
         }
 
         if (exprType.tag != TypeTags.UNION) {
-            if (types.isAssignable(exprType, symTable.errorType)) {
+            if (exprType.tag == TypeTags.READONLY) {
+                checkedExpr.equivalentErrorTypeList = new ArrayList<>(1) {{
+                    add(symTable.errorType);
+                }};
+                resultType = symTable.anyAndReadonly;
```

Comment:
Shouldn't we set the type for the `checkedExpr` here? In the union path, at L4711 we seems to be calling `types.checkType(checkedExpr, actualType, expType);`.

Reply:
The `resultType` will be set as expression type in here https://github.com/ballerina-platform/ballerina-lang/blob/92f603f72b754a3b34bb80ba852271a83a743948/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java#L374